stream/replicator.h

```diff
@@ -332,19 +377,35 @@ class SubscribableRemoteStream final {
   template <typename F>
   RemoteSubscriberScope<F> Subscribe(F& subscriber,
                                      uint64_t start_idx = 0u,
-                                     bool checked_subscription = false,
+                                     std::chrono::microseconds from_us = std::chrono::microseconds(0),
+                                     SubscriptionMode subscription_mode = SubscriptionMode::Unchecked,
                                      std::function<void()> done_callback = nullptr) const {
     static_assert(current::ss::IsStreamSubscriber<F, entry_t>::value, "");
-    return RemoteSubscriberScope<F>(stream_, subscriber, start_idx, checked_subscription, done_callback);
+    return RemoteSubscriberScope<F>(stream_, subscriber, start_idx, from_us, subscription_mode, done_callback);
+  }
+
+  template <typename F, ReplicationMode RM>
+  void FlipToMaster(F& subscriber, head_optidxts_t head_idxts, uint64_t key, SubscriptionMode subscription_mode) const {
+    static_assert(current::ss::IsStreamSubscriber<F, entry_t>::value, "");
+    const auto response = HTTP(GET(stream_->GetFlipToMasterURL(head_idxts, key, subscription_mode)));
+    if (response.code == HTTPResponseCode.OK) {
+      const auto start_idx = Exists(head_idxts.idxts) ? Value(head_idxts.idxts).index + 1 : 0;
+      RemoteStreamSubscriber<F, entry_t, RM> remote_subscriber(stream_, subscriber, start_idx);
```

Comment:
Nit: These two statements can be a one-liner, as all you need to do it call `.PassChunkToSubscriber()` once on a temporarily-created `RemoteStreamSubscriber<>`.